app/models/agents/csv_agent.rb

```diff
@@ -0,0 +1,194 @@
+module Agents
+  class CsvAgent < Agent
+    include FormConfigurable
+    include FileHandling
+
+    cannot_be_scheduled!
+
+    def default_options
+      {
+        'mode' => 'parse',
+        'separator' => ',',
+        'use_fields' => '',
+        'output' => 'event_per_row',
+        'with_header' => 'true',
+        'data_path' => '$.data',
+        'data_key' => 'data'
+      }
+    end
+
+    description do
+      <<-MD
+        The `CsvAgent` parses or serializes CSV data. When parsing events are emitted for every row or one event containing all rows.
+
+        `mode` determines if the agent is parsing CSV from incoming event serializes the data of events to CSV.
+
+        ### Universal options
+
+        Specify the `separator` which is used to seperate the fields from each other.
+
+        `data_key` sets in the key which contains the serialized CSV or parsed CSV data in the emitted event.
```

Comment:
Is this right? "`data_key` will be the key in the outgoing event that contains the CSV."